rundeckapp/src/main/groovy/com/dtolabs/rundeck/app/support/BuilderUtil.groovy

```diff
@@ -229,6 +229,17 @@ class BuilderUtil {
         os.replaceAll('(\r\n|\r|\n)', lineEnding)
     }
 
+    /**
+     * Trim all empty space from each line in the input string and
+     * replace all line endings with the given string
+     * @param os input string
+     * @param lineEnding line ending string to use
+     * @return new string
+     */
+    static String trimAllLinesAndReplaceLineEndings(String os, String lineEnding) {
+        os.readLines().collect { it.trim() }.join(lineEnding)
```

Comment:
String.trim removes both *leading* and *trailing* whitespace.  I think it should not remove leading whitespace, so `.trim()` is probably not the right method to use.  Test cases don't have any cases with leading whitespace.

Reply:
I've changed it to trim the end of the line and added tests.